### PR TITLE
Migrate from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by illicitonion**
Apparently we hit a usage limit with Travis.

Also, address clippy warnings.

### CODE REVIEW

File: .github/workflows/presubmit.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+name: presubmit
+on:
+  push:
```
Comment (danielhenrymantilla) [inline .github/workflows/presubmit.yml:3]:
```suggestion
  push:
    branches: ['main']
```

Let's keep a "lazy" CI on secondary branches; those would need to opt into CI by having a PR around them 🙂

File: num_enum_derive/src/lib.rs [hunk 2]
```diff
@@ -94,11 +94,15 @@ struct VariantAlternativesAttribute {
...
+        let expressions = content.parse_terminated(Expr::parse)?;
         Ok(Self {
-            keyword: input.parse()?,
-            _eq_token: input.parse()?,
-            _bracket_token: syn::bracketed!(content in input),
-            expressions: content.parse_terminated(Expr::parse)?,
+            keyword,
+            _eq_token,
+            _bracket_token,
+            expressions,
```
Comment (danielhenrymantilla) [inline num_enum_derive/src/lib.rs:105]: This is more verbose for no real good reason, imho; I guess it's just another instance of `clippy` being quite opinionated 🤷

**Review by danielhenrymantilla — APPROVED** [review]
LGTM ✅

File: num_enum_derive/src/lib.rs (hunk 2 above)
Comment (illicitonion) [reply num_enum_derive/src/lib.rs:105]: I buy that "initialisation order of struct fields isn't _super_ obvious" is a motivation, I don't feel particularly strongly either way but 🤷 I don't dislike it enough to work around it :)

**Pull request merged**